unittest_stralloc.c

```diff
@@ -0,0 +1,48 @@
+#include <check.h>
+
+#include "stralloc.h"
+
+START_TEST(test_stralloc_thingy)
+{
+  stralloc thingy = {0};
+  char *input = "thingy";
```

Comment:
const? Depends on if it causes a warning when being const, but should usually be a good idea.